core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSinkVersion2.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Eclipse RDF4J contributors, Aduna, and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.sail.base;
+
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.sail.SailException;
+
+/**
+ * An extension of the SailSink interface to support Statements directly for better performance.
+ *
+ * @author Håvard Ottestad
+ */
+public interface SailSinkVersion2 extends SailSink {
```

Comment:
I'd prefer adding the additional methods as default methods to the original interface instead.

Reply:
The default implementation would be to just redirect to the subject-predicate-object-context version of the method.